auth_oidc/models/auth_oauth_provider.py

```diff
@@ -52,8 +52,9 @@ def _get_key(self, kid):
         r.raise_for_status()
         response = r.json()
         for key in response["keys"]:
-            if key["kid"] == kid:
+            if key.get("kid", None) == kid:
                 return key
+
```

Comment:
I think it would be easier to read (by me at least :) if it was `if kid is None: return somekey else: return key["kid"]`.

And, do you have a standard reference that says which key should be used in that case? Here you take the "first" one. But why the first?

Reply:
Well, no i do not take the first one. I only take the key that has no `kid` specified. As this is only allowed when only one key is specified in the JWKS it should be also the first (and last) one.

Your proposed code would get rid of the check that if ID Token has no kid specified in header the JWKS also needs to have no kid specified for the key.

Reply:
As per my findings in https://github.com/OCA/server-auth/pull/436#discussion_r1015223912 i would roll back the roll back of the for loop (So we have a _get_keys() again).

And i would also change the code so an ID Token without a kid can match a single key in JWKS.

@sbidoul Let me know if this approach looks good to you.

Reply:
Yes, sounds good, thanks!